Odata-docs/odatalib/v7/using-ut8jsonwriter-for-better-performance.md

```diff
@@ -0,0 +1,58 @@
+
+Here's an example:
+
+```csharp
+using Microsoft.OData;
+using Microsoft.OData.Json;
+```
+
+```c#
+IContainerBuilder builder = new TestContainerBuilder();
+builder.AddDefaultODataServices();
+builder.AddService<IStreamBasedJsonWriterFactory>(sp => DefaultStreamBasedJsonWriterFactory.Default);
+```
+
+## Limitations and other considerations
+
+### Choosing a JavaScriptEncoder
+
+The `DefaultStreamBasedJsonWriterFactory.Default` instance uses the default [JavaScriptEncoder](/dotnet/api/system.text.encodings.web.javascriptencoder.default) used by `Utf8JsonWriter` to encode and escape output text. The default encoder escapes control characters, non-ASCII characters and HTML-sensitive characters like `<`. This differs from OData’s default behavior which does not escape HTML-sensitive characters.
+
+You can provide a different `JavaScriptEncoder`. Instead of injecting `DefaultStreamBasedJsonWriterFactory.Default`, you can create a new instance of the factory and pass the encoder to the constructor. The following snippet uses the [JavaScriptEncoder.UnsafeRelaxedJsonEscaping](/dotnet/api/system.text.encodings.web.javascriptencoder.unsaferelaxedjsonescaping) which does not escape HTML-sensitive characters. This is the default encoder used by ASP.NET Core.
+
+```c#
+IStreamBasedJsonWriterFactory factory = new DefaultStreamBasedJsonWriterFactory(JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+```
+
+Note that there’s no built-in JavaScriptEncoder that exactly matches the escaping rules of `Microsoft.OData.Core`’s default `JsonWriter`. The raw output of the two writers will be different. But if the client is [standards-compliant](https://www.ietf.org/rfc/rfc4627.txt), this should not be a problem. Nevertheless, this is something you should take into consideration before adopting the new writer. In general, you should not be highly dependent on a particular escaping output because the internal implementation may change from one version of .NET to another (See relevant discussions [here](https://github.com/dotnet/runtime/issues/70419), [here](https://github.com/dotnet/runtime/issues/54193) and [here](https://github.com/dotnet/runtime/issues/1564#issuecomment-504780719)).
+
+### Supported .NET versions
+
+This feature is only available with Microsoft.OData.Core on .NET Core 3.1 and above (i.e. .NET 6 etc.). It is not supported on Microsoft.OData.Core versions running on .NET Framework.
+
+### No support for streaming writes
+
+The default `JsonWriter` implements the `IJsonStreamWriter` and `IJsonStreamWriterAsync` interfaces. This interface allows you to pipe data directly from an input stream to the output destination. This may be useful when writing the contents of large binary file into base-64 output without buffering everything in memory first. The new `ODataUtf8JsonWriter` does not yet implement this interface, which means input from a stream will be buffered entirely in memory before being written out to the destination.
```

Comment:
```suggestion
The default `JsonWriter` implements the `IJsonStreamWriter` and `IJsonStreamWriterAsync` interfaces. This interface allows you to pipe data directly from an input stream to the output destination. This may be useful when writing the contents of large binary file into base-64 output without buffering everything in memory first. The new `ODataUtf8JsonWriter` does not yet implement this interface, which means input from a stream will be buffered entirely in memory before being written out to the destination. If your use case requires the streaming to a stream it is recommended to use the default `JsonWriter`.
```

Reply:
I've added the suggestion but rephrased it to: 

If your use case requires writing from a stream input, consider using the default `JsonWriter`.